packages/theme-default/src/components/Search/logic/search.ts

```diff
@@ -260,18 +265,22 @@ export class PageSearcher {
           length: getStrByteLength(query),
         },
       ];
-      matchedResult.push({
-        type: 'content',
-        title: item.title,
-        header: currentHeader?.text ?? item.title,
-        statement,
-        highlightInfoList,
-        link: `${domain}${normalizeHref(item.routePath)}${
-          currentHeader ? `#${currentHeader.id}` : ''
-        }`,
-        query,
-        group: this.#options.extractGroupName(item.routePath),
-      });
+      if (!currentHeader || !matchHeaderSet?.has(currentHeader)) {
```

Comment:
This if sentence seems no need since already return `if (currentHeader && matchHeaderSet?.has(currentHeader))`?

Reply:
return of `if  (currentHeader && matchHeaderSet?.has(currentHeader))` located at `queryIndex === -1` condition, this if sentence run in `queryIndex !== -1` condition.

Reply:
Get it, we can use `const hasHeader = currentHeader && matchHeaderSet?.has(currentHeader`, and use `hasHeader` and `!hasHeader`.

Reply:
Good idea, I add a function to check is header matched.